vulture_os/toolkit/api_parser/vectra/vectra.py

```diff
@@ -221,8 +221,9 @@ def execute(self):
 
             # And update lock after sending lines to Rsyslog
             self.update_lock()
-            self.frontend.last_api_call = to
-            self.frontend.save(update_fields=['last_api_call'])
+            if results:
+                self.frontend.last_api_call = datetime.fromisoformat(results[-1]["event_timestamp"].replace("Z", "+00:00")) + timedelta(milliseconds=1)
```

Comment:
are you sure events are ordered?
also: the `results` variable is an aggregation of multiple possible endpoints, so the last log may not be the "real" last!
also bis: you should probably take into account the possible drift of the collector to more than 24h in the past (and the update of the `last_api_call` to keep below the 24h delay)